python/tvm/autotvm/tuner/callback.py

```diff
@@ -178,3 +178,82 @@ def _callback(tuner, inputs, results):
             sys.stdout.flush()
 
     return _callback
+
+
+def visualize_progress(
+    idx, title="AutoTVM Progress", si_prefix="G", keep_open=False, live=True, out_path=None
+):
+    """Display tuning progress in graph
+
+    Parameters
+    ----------
+    idx: int
+        Index of the current task.
+    title: str
+        Specify the title of the matplotlib figure.
+    si_prefix: str
+        SI prefix for flops
+    keep_open: bool
+        Wait until the matplotlib window was closed by the user.
+    live: bool
+        If false, the graph is only written to the file specified in out_path.
+    out_path: str
+        Path where the graph image should be written (if defined).
+    """
+    import matplotlib.pyplot as plt
+
+    class _Context(object):
```

Comment:
i'm not sure you need a class here to store the vars. they should be captured into the closure below if you reference them from there